### PR TITLE
Use Arc<DB> in DeltaReader fro cheap cloning and add sync methods

### PR DESCRIPTION
**Pull request by citizen-stig**
So it can be integrated in SDK

### CODE REVIEW

File: src/cache/delta_reader.rs [hunk 1]
```diff
@@ -16,126 +16,153 @@ pub type SnapshotIterRange<'a> = btree_map::Range<'a, SchemaKey, Operation>;
 /// Read-only data provider that supports a list of snapshots on top of [`DB`].
 /// Maintains total ordering and respects uncommited deletions.
 /// Should not write to underlying [`DB`].
-#[derive(Debug)]
+#[derive(Debug, Clone)]
 pub struct DeltaReader {
     /// Set of not commited changes in chronological order.
     /// Meaning that the first snapshot in the vector is the oldest and the latest is the most recent.
     /// If keys are equal, the value from more recent snapshot is taken.
     snapshots: Vec<Arc<SchemaBatch>>,
```
Comment (neysofu) [inline src/cache/delta_reader.rs:24]: Should we wrap `snapshots` in an `Arc` as well? Something like `snapshots: Arc<[Arc<SchemaBatch>]>`. Hard to say how expensive the clone will be without benchmarking though so maybe it's fine.

File: src/test.rs [hunk 2]
```diff
@@ -97,6 +97,12 @@ impl<S: Schema> KeyEncoder<S> for TestField {
     }
 }
 
+impl<S: Schema> SeekKeyEncoder<S> for TestField {
+    fn encode_seek_key(&self) -> Result<Vec<u8>, CodecError> {
+        Ok(self.0.to_be_bytes().to_vec())
```
Comment (neysofu) [inline src/test.rs:102]: nit: `Ok(self.as_bytes())` would avoid error-prone logic duplication.

**Review by neysofu — APPROVED** [review]
Left two minor comments, LGTM!

File: src/cache/delta_reader.rs (hunk 1 above)
Comment (citizen-stig) [reply src/cache/delta_reader.rs:24]: From "application logic" point of view length of this vector is going to be equal to number of non-finalized blocks. 
I don't think it is suppose be more than 50 blocks in the worst case, so I deliberately kept it as regular vector. 

I've added an item to https://github.com/Sovereign-Labs/sovereign-sdk-wip/issues/575 to add benchmarks there

Comment (neysofu) [reply src/cache/delta_reader.rs:24]: Alright!

**Pull request merged**